### ISSUE TITLE
OpenMPI 4.1.7 and 5.0.5 Incomplete Installations

### ISSUE THREAD

**Issue by ParticleTruthSeeker** [open-mpi/ompi#12917]
The builds wont produce openshmem wrapper and 5.0.5 links against wrong libopen-pal 40 vs 80 and still fails to produce the openshmem compiler wrappers.

Please submit all the information below so that we can understand the working environment that is the context for your question.

* If you have a problem building or installing Open MPI, [be sure to read this](https://docs.open-mpi.org/en/main/getting-help.html#for-problems-building-or-installing-open-mpi).
* If you have a problem launching MPI or OpenSHMEM applications, [be sure to read this](https://docs.open-mpi.org/en/main/getting-help.html#for-problems-launching-mpi-or-openshmem-applications).
* If you have a problem running MPI or OpenSHMEM applications (i.e., after launching them), [be sure to read this](https://docs.open-mpi.org/en/main/getting-help.html#for-problems-running-mpi-or-openshmem-applications).

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)
tar balls

### Please describe the system on which you are running

* Operating system/version: Debian 12
* Computer hardware: Ryzen 3
* Network type: Local

-----------------------------

## Details of the problem

Please describe, in detail, the problem that you are having, including the behavior you expect to see, the actual behavior that you are seeing, steps to reproduce the problem, etc.  It is most helpful if you can attach a small program that a developer can use to reproduce your problem.

**Note**: If you include verbatim output (or a code block), please use a [GitHub Markdown](https://help.github.com/articles/creating-and-highlighting-code-blocks/) code block like below:
```shell
shell$ mpirun -n 2 ./hello_world
```

**Comment by ParticleTruthSeeker** [comment]
I dont know whats wrong. These are my configuration arguments:
../openmpi-5.0.5/configure --with-hwloc=internal --with-libevent=internal --with-pmix=internal --with-prrte=internal CFLAGS=-m64 FCFLAGS=-m64 --with-wrapper-cflags=-m64 --with-wrapper-cxxflags=-m64 --with-wrapper-fcflags=-m64 --prefix=/home/user/OpenMPI/Install --with-openib --without-udapi --enable-mca-dso --with-slurm --disable-shared --enable-static

OpenMPI will not build with external  dependencies are they are too old but it still randomly links to system librarys. The system library itself will not build succesfully the examples or other MPI enabled programs. MPICH seems to work.

**Comment by jsquyres** [comment]
Can you submit the information requested here: https://docs.open-mpi.org/en/main/getting-help.html#for-problems-building-or-installing-open-mpi

**Comment by ParticleTruthSeeker** [comment]
> Can you submit the information requested here: https://docs.open-mpi.org/en/main/getting-help.html#for-problems-building-or-installing-open-mpi

Sure but I think the errors come from oshmem.
I am using Debian 12. On ubuntu24 5.0.5 built fine.
--- 
The associated compiler wrappers wheren't being built but no obvious configuration error appeared until I forced the configuration of openshmem with --enable-openshmem. This yielded an error concerning an old version of of UCX so I built UCX force the use of my build v1.17.0 and now the wrappers appeared and the examples seem to have been built.
The configure script for ucx and openmpi picks up random libraries from my path including from a Go installation which is something I would have hoped was controllable.

All the examples seem to run now. Having said that HYPRE examples won't run when bigint is enabled.

This is my configuration line.

../openmpi-5.0.5/configure --with-hwloc=internal --with-libevent=internal --with-pmix=internal --with-prrte=internal CFLAGS=-m64 FCFLAGS=-m64 --with-wrapper-cflags=-m64 --with-wrapper-cxxflags=-m64 --with-wrapper-fcflags=-m64 --prefix=/home/user/OpenMPI/Install --with-openib --without-udapi --enable-mca-dso --with-slurm --enable-oshmem --with-ucx=/home/user/OpenMPI/UCX/Install --enable-mca-no-build=btl-uct
